### PR TITLE
2nd part of stabilizing flaky websocket integration test

### PR DESCRIPTION
**Pull request by leo-da**
handling empty events, which we may receive a bunch in the end of the run when there is no activity

Closes: #4979

` flaky = True` does not always help, there were windows builds that failed all 3 flaky attempts.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by leo-da** [comment]
passed once

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala [hunk 1]
```diff
@@ -144,6 +157,14 @@ object domain {
 
     def toLedgerApi(o: Offset): lav1.ledger_offset.LedgerOffset =
       lav1.ledger_offset.LedgerOffset(lav1.ledger_offset.LedgerOffset.Value.Absolute(unwrap(o)))
+
+    implicit val ordering: Ordering[Offset] =
+      Ordering.by(LedgerOffsetUtil.parseOffsetString _ compose unwrap)(
+        LedgerOffsetUtil.LongEitherLongLongOrdering)
+
+    implicit val semigroup: Semigroup[Offset] = new Semigroup[Offset] {
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala]: "Later in the stream" is always the desired one; comparing them is an unneeded complication.

```suggestion
    implicit val semigroup: Semigroup[Offset] = Tag.unsubst(Semigroup[Offset @@ Tags.LastVal])
```

**Comment by leo-da** [comment]
2 runs passed

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala (hunk 1 above)
Comment (leo-da) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala]: `Tags.LastVal` is not the same as `Ordering.max`... why LastVal? We care about the largest offset, not the last one?

Comment (S11001001) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala]: No, in this case, we care about the last one.

We care about the largest one in the only temporally-unordered scenario: that is, the one that you originally implemented the comparison for. Stream steps are already temporally ordered, so introducing another layer of ordering on top of that is needless complication.

Comment (leo-da) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala]: plus it will break the associativity law.

Comment (S11001001) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala]: No, both max and lastval are associative. That is why scalaz defines semigroups for them at all.

Comment (leo-da) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala]: a1 + (a2 + a3) = a3
(a1 + a2) + a3 = a3
scratch the associaitivity law :)

Comment (leo-da) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala]: @S11001001 thanks... you are 100% right.... if I leave the semigroup implementation as ordering.max, the property that we observe in the test will always be true:
```
lastSeen should be > liveStart
```
so, yeah we should capture the last one, the left operand.

**Comment by leo-da** [comment]
3 consecutive runs passed without errors.

**Pull request merged**